lib/asciidoctor/parser.rb

```diff
@@ -1310,7 +1313,9 @@ def self.next_list_item(reader, list_block, match, sibling_trait = nil)
 
     # first skip the line with the marker / term
     reader.advance
-    list_item_reader = Reader.new read_lines_for_list_item(reader, list_type, sibling_trait, has_text), reader.cursor
+    cursor = reader.cursor
+
+    list_item_reader = Reader.new read_lines_for_list_item(reader, list_type, sibling_trait, has_text), cursor
```

Comment:
@mojavelinux Here's the trick, we are "saving" the cursor, otherwise the lineno is wrong on nested lists.